api/clients/codecs/default_blob_encoding_codec.go

```diff
@@ -0,0 +1,52 @@
+package codecs
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/encoding/utils/codec"
+)
+
+type DefaultBlobEncodingCodec struct{}
+
+var _ BlobCodec = DefaultBlobEncodingCodec{}
+
+func (v DefaultBlobEncodingCodec) EncodeBlob(rawData []byte) ([]byte, error) {
+	// encode blob encoding version byte
+	codecBlobHeader := EncodeCodecBlobHeader(byte(DefaultBlobEncoding), uint32(len(rawData)))
+
+	// append raw data
+	encodedData := append(codecBlobHeader, rawData...)
+
+	// encode all data modulo bn254
+	encodedData = codec.ConvertByPaddingEmptyByte(encodedData)
+
+	return encodedData, nil
+}
+
+func (v DefaultBlobEncodingCodec) DecodeBlob(encodedData []byte) ([]byte, error) {
+	// decode raw data modulo bn254
+	decodedData := codec.RemoveEmptyByteFromPaddedBytes(encodedData)
```

Comment:
I think we should only apply this to the header first, then read the header, then use the version byte in the header to determine what kind of padding decoding to apply to the rest of the blob. Otherwise the version byte doesn't really give any information on the encoding that's not determined optimistically.

Reply:
This all makes sense 👍